tiledb/sm/array/array.cc

```diff
@@ -1560,5 +1560,157 @@ void Array::set_serialized_array_open() {
   is_open_ = true;
 }
 
+void Array::get_max_tile_size(
+    tiledb_fragment_max_tile_sizes_t* max_tile_sizes) {
+  assert(max_tile_sizes);
+
+  if (remote_) {
+    auto rest_client = storage_manager_->rest_client();
+    if (rest_client == nullptr) {
+      throw StatusException(
+          Status_Error("Failed to obtain maximum tile size; remote array with "
+                       "no REST client."));
+    }
+
+    throw_if_not_ok(rest_client->get_array_maximum_tile_size_from_rest(
+        array_uri_, &max_tile_sizes->max_in_memory_tile_size));
+    return;
+  }
+
+  // Check that array is open
+  if (!is_open()) {
+    throw StatusException(Status_ArrayError(
+        "[Array::array_get_fragment_tile_max_size] Array is closed"));
+  }
+
+  if (query_type_ != QueryType::READ) {
+    throw StatusException(Status_ArrayError(
+        "[Array::array_get_fragment_tile_max_size] needs QueryType::READ"));
+  }
+
+  memset(max_tile_sizes, 0, sizeof(*max_tile_sizes));
+
+  EncryptionKey enc_key(config_);
+  uint64_t timestamp_start = 0;
+  uint64_t timestamp_end = UINT64_MAX;
+
+  // Create an ArrayDirectory object and then load the array schemas
+  // and fragment metadata.
+  auto array_dir =
+      ArrayDirectory(resources_, array_uri_, timestamp_start, timestamp_end);
+
+  MemoryTracker memory_tracker;
```

Comment:
We could set a budget here for the tracker to sm.mem.total_budget.

Reply:
done